chains/runtime-common/src/processors/v2/fallback_message_processor.rs

```diff
@@ -0,0 +1,234 @@
+    )>,
+);
+
+impl<
+        T,
+        AccountId,
+        GatewayAddress,
+        DefaultClaimer,
+        EthereumNetwork,
+        EthereumUniversalLocation,
+        TanssiUniversalLocation,
+        XcmProcessor,
+        XcmWeigher,
+    > FallbackMessageProcessor<AccountId>
+    for AssetTrapFallbackProcessor<
+        T,
+        GatewayAddress,
+        DefaultClaimer,
+        EthereumNetwork,
+        EthereumUniversalLocation,
+        TanssiUniversalLocation,
+        XcmProcessor,
+        XcmWeigher,
+    >
+where
+    T: snowbridge_pallet_inbound_queue::Config
+        + pallet_xcm::Config
+        + snowbridge_pallet_system::Config,
+    [u8; 32]: From<<T as frame_system::Config>::AccountId>,
+    GatewayAddress: Get<H160>,
+    DefaultClaimer: Get<<T as frame_system::Config>::AccountId>,
+    EthereumNetwork: Get<NetworkId>,
+    EthereumUniversalLocation: Get<InteriorLocation>,
+    TanssiUniversalLocation: Get<InteriorLocation>,
+    XcmProcessor: ExecuteXcm<<T as pallet_xcm::Config>::RuntimeCall>,
+    XcmWeigher: WeightBounds<<T as pallet_xcm::Config>::RuntimeCall>,
+{
+    fn handle_message(
+        _who: AccountId,
+        message: Message,
+    ) -> Result<[u8; 32], MessageProcessorError> {
+        let extracted_message: ExtractedXcmConstructionInfo<
+            <T as pallet_xcm::Config>::RuntimeCall,
+        > = ExtractedXcmConstructionInfo {
+            origin: message.origin,
+            maybe_claimer: message.claimer.clone(),
+            assets: message.assets.clone(),
+            eth_value: message.value,
+            execution_fee_in_eth: message.execution_fee,
+            nonce: message.nonce,
+            user_xcm: Xcm::new(),
+        };
+
+        let prepared_xcm = crate::processors::v2::prepare_raw_message_xcm_instructions::<T>(
+            EthereumNetwork::get(),
+            &EthereumUniversalLocation::get(),
+            &TanssiUniversalLocation::get(),
+            GatewayAddress::get(),
+            DefaultClaimer::get(),
+            crate::processors::v2::RAW_MESSAGE_PROCESSOR_TOPIC_PREFIX,
+            extracted_message,
+        )
+        .map_err(|asset_derivation_error| {
+            MessageProcessorError::ProcessMessage(DispatchError::Other(
+                format!(
+                    "Error while preparing xcm instructions: {:?}",
+                    asset_derivation_error
+                )
+                .leak(),
+            ))
+        })?
+        .into();
+
+        let eth_location_reanchored_to_tanssi = reanchor_location_to_tanssi(
+            &EthereumUniversalLocation::get(),
+            &TanssiUniversalLocation::get(),
+            ().into(),
+        )
+        .map_err(|asset_derivation_error| {
+            MessageProcessorError::ProcessMessage(DispatchError::Other(
+                format!(
+                    "Unable to reanchor eth location to tanssi: {:?}",
+                    asset_derivation_error
+                )
+                .leak(),
```

Comment:
I see multiple uses of `String::leak`, is this copied from polkadot-sdk? In normal software you don't want to leak memory, but maybe if this is runtime code we don't care?

Reply:
I don't think this error message is needed. Ask auditors whether leaking memory here is ok or not.